src/metaswitch/sasclient/sender.py

```diff
@@ -54,27 +55,36 @@ def run(self):
         If the queue has been terminated (via _stopper), then stop.
         Maintains the connection - if the connection is down then reconnect using connect()
         """
-        while not self._stopper.is_set():
-            if not self._connected:
-                # Try to reconnect and have another go at the loop.
-                self.reconnect()
-                continue
-
-            # Try to get a message off of the queue.  If there's nothing there after a second,
-            # send a heartbeat.
-            try:
-                message = self._queue.get(True, 1)
-                self._queue.task_done()
-            except Queue.Empty:
-                message = messages.Heartbeat()
-
-            # Send the message.  If we fail, we'll want to reconnect.
-            if not self.send_message(message):
-                logger.debug("Failed to send a message, flag that we need to reconnect")
-                self._connected = False
-
-        self.disconnect()
-        self._connected = False
+        try:
+            while not self._stopper.is_set():
+                if not self._connected:
+                    # Try to reconnect and have another go at the loop.
+                    self.reconnect()
+                    continue
+
+                # Try to get a message off of the queue.  If there's nothing there after a second,
+                # send a heartbeat.
+                try:
+                    message = self._queue.get(True, 1)
+                    self._queue.task_done()
+                except Queue.Empty:
+                    message = messages.Heartbeat()
+
+                # Send the message.  If we fail, we'll want to reconnect.
+                if not self.send_message(message):
+                    logger.debug("Failed to send a message, flag that we need to reconnect")
+                    self._connected = False
+
+            self.disconnect()
+            self._connected = False
+
+        except Exception as e:
+            details = traceback.format_exc()
+            error = ("ERROR - Hit exception in SAS sender thread!  SAS logs will no longer be "
+                    "made.\n{}".format(details))
+            print(error)
```

Comment:
I think you're right about it being helpful to print the error to stdout, but you should probably add a comment to explain your reasoning so that someone doesn't come along and think that it's a mistake later!